crates/ruff_linter/src/checkers/ast/analyze/definitions.rs

```diff
@@ -184,14 +182,7 @@ pub(crate) fn definitions(checker: &mut Checker) {
                 continue;
             };
 
-            let contents = checker.locator().slice(string_literal);
-
-            let indentation = checker.locator().slice(TextRange::new(
-                checker.locator.line_start(string_literal.start()),
-                string_literal.start(),
-            ));
-
-            if string_literal.value.is_implicit_concatenated() {
+            let [sole_string_part] = string_literal.value.as_slice() else {
```

Comment:
I found the old explicit implicit concatenated string helped with readability. I wonder if we should add a method for non-implicit concatenated strings to `string_literal`. 

Coming up with a good name is hard: `let Some(sole_string_part) = string_literal.as_non_implicit()`

Reply:
Yeah, good point about readability. I will defer this to another PR, though, as I think almost every usage of the `StringLiteralValue::as_slice()` method would probably switch to using this new API -- and I agree that what we would call it isn't necessarily obvious. It seems worthy of considering as its own thing.

For now I'll just add a comment here.